src/System.Private.CoreLib/shared/System/Threading/SynchronizationContext.cs

```diff
@@ -0,0 +1,64 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Thread = Internal.Runtime.Augments.RuntimeThread;
+
+namespace System.Threading
+{
+    public partial class SynchronizationContext
+    {
+        private bool _requireWaitNotification;
+
+        public SynchronizationContext()
+        {
+        }
+
+#if !FEATURE_APPX
+        public static SynchronizationContext Current => Thread.CurrentThread.SynchronizationContext;
+#endif
+
+        protected void SetWaitNotificationRequired() => _requireWaitNotification = true;
+
+        public bool IsWaitNotificationRequired() => _requireWaitNotification;
+
+        public virtual void Send(SendOrPostCallback d, object state) => d(state);
+
+        public virtual void Post(SendOrPostCallback d, object state) => ThreadPool.QueueUserWorkItem(s => s.d(s.state), (d, state), preferLocal: false);
+
+        /// <summary>
+        ///     Optional override for subclasses, for responding to notification that operation is starting.
+        /// </summary>
+        public virtual void OperationStarted()
+        {
+        }
+
+        /// <summary>
+        ///     Optional override for subclasses, for responding to notification that operation has completed.
+        /// </summary>
+        public virtual void OperationCompleted()
+        {
+        }
+
+        [CLSCompliant(false)]
+        public virtual int Wait(IntPtr[] waitHandles, bool waitAll, int millisecondsTimeout)
+        {
+            return WaitHelper(waitHandles, waitAll, millisecondsTimeout);
+        }
+
+        [CLSCompliant(false)]
+        protected static int WaitHelper(IntPtr[] waitHandles, bool waitAll, int millisecondsTimeout)
+        {
+            if (waitHandles == null)
+            {
+                throw new ArgumentNullException(nameof(waitHandles));
+            }
+
+            return WaitHelperNative(waitHandles, waitAll, millisecondsTimeout);
```

Comment:
This method logically belongs to WaitHandle. Forwarding this onto a method on WaitHandle like it is done in CoreRT seems better to me.

Reply:
(Maybe even keep the null check for the argument in the WaitHandle part.)